src/plugins/formulas/engine/register.js

```diff
@@ -0,0 +1,192 @@
+import staticRegister from '../../../utils/staticRegister';
+import { isUndefined } from '../../../helpers/mixed';
+import { warn } from '../../../helpers/console';
+import { PLUGIN_KEY } from '../formulas';
+import { DEFAULT_LICENSE_KEY, getEngineSettingsWithDefaultsAndOverrides } from './settings';
+
+/**
+ * Setups the engine instance. It either creates a new (possibly shared) engine instance, or attaches
+ * the plugin to an already-existing instance.
+ *
+ * @param {object} hotSettings Object containing the Handsontable settings.
+ * @param {string} hotId Handsontable guid.
+ * @returns {null|object} Returns the engine instance if everything worked right and `null` otherwise.
+ */
+export function setupEngine(hotSettings, hotId) {
+  const pluginSettings = hotSettings[PLUGIN_KEY];
+  const engineConfigItem = pluginSettings.engine;
+
+  if (isUndefined(engineConfigItem)) {
+    warn('Missing the required `engine` key in the Formulas settings. Please fill it with either an' +
+      ' engine class or an engine instance.');
+
+    return null;
+  }
+
+  // `engine.hyperformula` or `engine` is the engine class
+  if (typeof engineConfigItem.hyperformula === 'function' || typeof engineConfigItem === 'function') {
+    return registerEngine(
+      engineConfigItem.hyperformula ?? engineConfigItem,
```

Comment:
There's a reason that we want to support an alternative way of passing the engine class?
The standard way:
```js
      formulas: {
        engine: HyperFormula,
      },
```
An alternative way:
```js
      formulas: {
        engine: {
          hyperformula: HyperFormula
        },
      },
```
With an alternative way, we could pass the specified config option directly to the HF. Is that it?

Reply:
I can respond to myself 😄  I've checked that and yes. The alternative way can be used to customize the HF engine.

Reply:
@budnix 🤝 @budnix